### PR TITLE
Use esbuild directly, without estrella

### PR DESCRIPTION
**Pull request by jimrandomh**
Previously, we were using a library called `estrella`, which is a wrapper around esbuild. Estrella starts and restarts a server, runs `tsc`, and does command-line argument parsing, but in exchange for these small bits of funcitonality it kept us back on an outdated version of esbuild.

This PR removes Estrella, replacing all its functionality with doing stuff directly in our build script. It also converts our build script from JS to TS.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208670909081755) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by jimrandomh** [comment]
Note: The build script has a lot of entry points with a lot of different command line arguments, and they can't all be tested locally. Make sure that a staging server of some sort is able to start successfully, before merging this to master.

File: build.ts [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+  e2e: false,
+  settings: null,
+  db: null,
+  postgresUrl: null,
+  postgresUrlFile: null,
+  shell: false,
+  command: null,
+  lint: false,
+}
+const helpText = (argv0: string) => `usage: yarn ts-node build-esbuild.ts [options]`
```
Comment (b0b3rt) [inline build.ts:39]: Is this used anywhere?  (Also, the argument appears to be unused.)

File: build.ts [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+
+let latestCompletedBuildId: string|null = generateBuildId();
+let inProgressBuildId: string|null = null;
+const serverPort = getServerPort();
+const websocketPort = serverPort + 1;
+
+setOutputDir(`./build${serverPort === defaultServerPort ? "" : serverPort}`);
+
+// Two things this script should do, that it currently doesn't:
+//  * Provide a websocket server for signaling autorefresh
```
Comment (b0b3rt) [inline build.ts:138]: Is this still missing?  (Also, was there a second thing that it should do that it currently doesn't?)

File: build.ts [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+
+const isProduction = !!opts.production;
+const isE2E = !!opts.e2e;
+const settingsFile = opts.settings || "settings.json"
+
+// Allow FM_WATCH to override the --watch CLI flag that is passed in
+/*const watchEnvVar = process.env.FM_WATCH?.toLowerCase();
+if (watchEnvVar === 'true' || watchEnvVar === 'false') {
+  cliopts.watch = watchEnvVar === 'true';
+}*/
```
Comment (b0b3rt) [inline build.ts:148]: Does this need to be uncommented/reimplemented?

File: build.ts [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+}
+
+const serverBundleDefinitions: Record<string,string> = {
+  "bundleIsServer": "true",
+  "buildProcessPid": `${process.pid}`,
+}
+
+class RunningServer {
+  private argv: string[];
+  private pid: number | null = null;
```
Comment (oetherington) [inline build.ts:197]: `this.pid` is written to, but never read

**Review by oetherington — APPROVED** [review]
LGTM

Maybe we should clear the console when rebuilding like estrella does? I don't have a strong opinion but it's something to consider.

**Pull request merged**